File: 118_pascal_triangle/try.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector< vector<int> > ans;
        for(int i = 0; i < numRows; ++i){
            vector<int>row;
            if(i == 0) row.push_back(1);
            else{
                for(int j = 0; j < i+1; ++j){
                    int element = (j == 0 || j == i ? 1: ans[i-1][j-1]+ans[i-1][j]);
                    row.push_back(element);
                }
            }
            ans.push_back(row);
        }
        return ans;
    }
};

File: 118_pascal_triangle/try.py
class Solution:
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        """
        Complexity: O(numRows^2), 1+2+3+...+numRows
        Space: O(numRows^2)
        """
        ans = []
        for i in range(numRows):
            if i == 0:
                ans.append([1])
                continue
            row = [1 if j == 0 or j == i else ans[i-1][j-1] + ans[i-1][j] for j in range(i+1)]
            ans.append(row)
        return ans
